libpod/container_internal_linux.go

```diff
@@ -698,11 +702,37 @@ func (c *Container) makeBindMounts() error {
 			// If it doesn't, don't copy them
 			resolvPath, exists := bindMounts["/etc/resolv.conf"]
 			if exists {
-
 				c.state.BindMounts["/etc/resolv.conf"] = resolvPath
 			}
+
+			// check if dependency container has an /etc/hosts file
 			hostsPath, exists := bindMounts["/etc/hosts"]
-			if exists {
+
+			// if not, and the new container has hosts to add, set the hosts basis path to /etc/hosts
+			if !exists && (len(c.config.HostAdd) > 0 || (len(c.state.NetworkStatus) > 0 && len(c.state.NetworkStatus[0].IPs) > 0)) {
+				hostsPath = defaultHostFile
+			}
+			// only go through all of this if we want to add a hosts file
+			if len(hostsPath) > 0 {
+				depCtr.lock.Lock()
+				// generate a hosts file for the dependency container,
+				// based on either its old hosts file, or the default,
+				// and add the relevant information from the new container (hosts and IP)
+				hostsPath, err = depCtr.generateHosts(hostsPath, c)
```

Comment:
I dunno if I like this... We won't pick this up until a restart, so it's not really shared.

Do we need to force sharing of /etc/hosts in this case, or is it sufficient to create it just for this container? We override resolv.conf for custom DNS settings when sharing network namespaces.

Reply:
How would other containers that are added to the namespace see it if a container they don't depend on (because they are laterally equal in the dependency tree)? 

My other question is: where besides /etc/host creation is the BindMount() map needed? If a dependency container was started up and had an empty /etc/hosts, but needed it for whatever process that was running (that shouldn't be restarted to see these changes), wouldn't that process see the changes when the new container appended to the depCtr's /etc/hosts file? I guess the question is: the state of a process that runs will be out of sync, but only on the libpod end and not from the perspective of namespace management, which I think would work, right?

Reply:
We use the map to determine what bind mounts to add when creating the container spec. If we add to it after container creation time, it will never be picked up before a restart. Actually, now that I think about it, we force a regeneration of the bind mounts on each restart, so we'll never pick up this change - we'll overwrite it on restart.